tests/Common/CommonSchemaTest.php

```diff
@@ -405,6 +406,30 @@ public function testGetTableNames(array $pdoAttributes): void
         $db->close();
     }
 
+    public function testHasTable(): void
+    {
+        $db = $this->getConnection(true);
+
+        $schema = $db->getSchema();
+
+        $tempTableName = 'testTemporaryTable';
+        $db->createCommand()->createTable($tempTableName, [
+            'id' => ColumnBuilder::primaryKey(),
+            'name' => ColumnBuilder::string()->notNull(),
+        ])->execute();
+
+        $this->assertTrue($schema->hasTable('order'));
+        $this->assertTrue($schema->hasTable('category'));
+        $this->assertTrue($schema->hasTable($tempTableName));
+        $this->assertFalse($schema->hasTable('no_such_table'));
+
+        $db->createCommand()->dropTable($tempTableName)->execute();
+
+        $this->assertFalse($schema->hasTable($tempTableName, '', true));
```

Comment:
```suggestion
        $db->createCommand('DROPT TABLE ' . $db->getQuoter()->quoteTableName($tempTableName))->execute();

        $this->assertTrue($schema->hasTable($tempTableName));
        $this->assertFalse($schema->hasTable($tempTableName, '', true));
```

We check that the table still exists after deletion, but is missing after cache reset.